base_rest/http.py

```diff
@@ -172,13 +172,19 @@ def _handle_exception(self, exception):
         try:
             return super(HttpRestRequest, self)._handle_exception(exception)
         except (UserError, ValidationError) as e:
-            return wrapJsonException(BadRequest(e.name), include_description=True)
+            extra_info = getattr(e, "rest_json_info", None)
```

Comment:
@simahawk To avoid code redundancy we could put at the start of the `wrapJsonException`method:
```python
def wrapJsonException(exception, include_description=False, extra_info=None):
    """Wrap exceptions to be rendered as JSON.
    :param exception: an instance of an exception
    :param include_description: include full description in payload
    :param extra_info: dict to provide extra keys to include in payload
    """
    extra_info = extra_info or getattr(exception, "rest_json_info", None)
    
``` 
What do you think about?

Reply:
you should pass the original exception there (I thought about it, decided to keep it simple) because `exception` in this case is the wrapping one.

Reply:
indeed....